doc/api/esm.md

```diff
@@ -1160,29 +1160,30 @@ condition list **must** be passed through to the `defaultResolve` function.
 ```js
 /**
  * @param {string} specifier
- * @param {object} context
- * @param {string} context.parentURL
- * @param {string[]} context.conditions
- * @param {function} defaultResolve
- * @returns {object} response
- * @returns {string} response.url
+ * @param {{
+ *   parentURL:!(string|undefined),
+ *   conditions:Array<string>,
+ * }} context
+ * @param {Function} defaultResolve
+ * @returns {{url:string}}
  */
-export async function resolve(specifier, context, defaultResolve) {
+export function resolve(specifier, context, defaultResolve) {
   const { parentURL = null } = context;
-  if (someCondition) {
+  if (Math.random()) { // Some condition.
```

Comment:
Is this necessary? If we're going to use this it should be `Math.random() > 0.5`, so that it's not always true; but why is just `someCondition` a problem?

Reply:
`someCondition` is `undefined`, the compilers don't know what to do here.

Reply:
Sure, but this makes the example harder to understand in my opinion. Why do compilers need to be able to compile this at all? Isn't the point of fixing the signature so that users can copy/paste this example and then replace the body with their own?

Reply:
> Why do compilers need to be able to compile this at all?

The way to type checking is performed by compilers requires that the source code be valid.

Reply:
```suggestion
  if (Math.random() > 0.5) { // Some condition.
```